src/dev/i18n/extractors/html.js

```diff
@@ -140,54 +130,37 @@ function trimOneTimeBindingOperator(string) {
   return string;
 }
 
-/**
- * Remove interpolation expressions from angular and throw on `| i18n:` substring.
- *
- * Correct usage: `<p aria-label="{{ ::'namespace.id' | i18n: { defaultMessage: 'Message' } }}"></p>`.
- *
- * Incorrect usage: `ng-options="mode as ('metricVis.colorModes.' + mode | i18n: { defaultMessage: mode }) for mode in collections.metricColorMode"`
- *
- * @param {string} string html content
- */
-function validateI18nFilterUsage(string) {
-  const stringWithoutExpressions = string.replace(ANGULAR_EXPRESSION_REGEX, '');
-  const i18nMarkerPosition = stringWithoutExpressions.indexOf(I18N_FILTER_MARKER);
-
-  if (i18nMarkerPosition === -1) {
-    return;
-  }
-
-  const linesCount = (stringWithoutExpressions.slice(0, i18nMarkerPosition).match(/\n/g) || [])
-    .length;
-
-  const errorWithContext = createParserErrorMessage(string, {
-    loc: {
-      line: linesCount + 1,
-      column: 0,
-    },
-    message: 'I18n filter can be used only in interpolation expressions',
-  });
-
-  throw createFailError(errorWithContext);
-}
-
 function* getFilterMessages(htmlContent) {
-  validateI18nFilterUsage(htmlContent);
+  const allElements = cheerio.load(htmlContent)('*');
```

Comment:
__suggestion:__ can we reduce the number of map, filter etc. on same or similara data sets and move it to a dedicated function? Something like this (semi-pseudo code):
```js
function* extractExpressions(htmlContent) {
  for (const element of cheerio.load(htmlContent)('*').toArray()) {
    // Extract this....
    for (const node of element.children) {
      if (node.type === 'text') {
        yield* (node.data.match(ANGULAR_EXPRESSION_REGEX) || [])
          .filter(exp => exp.includes(I18N_FILTER_MARKER))
          .map(trimCurlyBraces);
      }
    }

    // Extract that....
    for (const attribute of Object.values(element.attribs)) {
      if (attribute.includes(I18N_FILTER_MARKER)) {
        yield trimCurlyBraces(attribute);
      }
    }
  }
}

function* getFilterMessages(htmlContent) {
  for (const expression of extractExpressions(htmlContent)) {
  .....
```